dustinsgoodman/next-playground
# Create Github User Search UI

# Overview

Creates a custom user search interface using the Github GraphQL API.

# Screenshots for Mobile and Desktop views (if applicable)

## Before
N/A

## After
![image](https://user-images.githubusercontent.com/1815379/102004330-b2321e00-3cdd-11eb-9705-40c54e8bf7f9.png)


# Details

## General
- [x] user can search against the API and get paginated results
- [x] user can see total count of results back from request
- [x] user can paginate forward and backward 1 page at a time
- [x] useful user information for results is displayed
- [x] clicking on a result takes you to the profile on Github.com


## Automated Testing
No tests currently exist. This PR is un-mergeable until I can get them done. Started running low on time.

### Manual Testing
If you want to make a Github key:
1. Generate a github access token and add to your `.env.local` as `NEXT_PUBLIC_GITHUB_ACCESS_TOKEN`
2. Run `yarn dev`
Go to 3 below

If you don't want to make a Github key:
1. Visit https://next-playground-5qmh6tfyd.vercel.app/
2. In the search bar, enter a username, e.g. `dustin`
Go to 3 below

--- 
3. See results load
4. Click "next" 2 times and see results load
5. Click "previous" 2 times and see previous results load
6. Click on any row and see Github.com open in new window
